workflows/templates/biascorrectdownscale.yaml

```diff
@@ -480,15 +506,24 @@ spec:
                   value: "{{ inputs.parameters.qdm-kind }}"
                 - name: out-zarr
                   value: "{{ tasks.get-output-downscaled-url.outputs.parameters.out-url }}"
-          - name: validate-downscaled
+          - name: apply-non-polar-dtr-ceiling-on-downscaled
+            template: apply-non-polar-dtr-ceiling
             depends: qplad
+            arguments:
+              parameters:
+                - name: in-zarr
+                  value: "{{ tasks.qplad.outputs.parameters.out-zarr }}"
+                - name: out-zarr
+                  value: "{{ tasks.qplad.outputs.parameters.out-zarr }}" # modifies input file !
```

Comment:
As noted above, I've got a feeling that you won't be able to read/write from the same Zarr Store without risking side-effects.